cves/CVE-2014-0231.yml

```diff
@@ -177,38 +207,43 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies:
-    note:
+    applies: true
+    note: |
+        timeouts are a layer of robustness underneath normal CGI input validation
   least_privilege:
-    applies:
+    applies: false
     note:
   frameworks_are_optional:
-    applies:
+    applies: false
     note:
   native_wrappers:
-    applies:
+    applies: false
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true 
+    note: |
+      centered around handling abnormal input (input which causes the script 
+      to halt)
   security_by_obscurity:
-    applies:
+    applies: false
     note:
   serial_killer:
-    applies:
+    applies: false
     note:
   environment_variables:
-    applies:
+    applies: false
     note:
   secure_by_default:
-    applies:
+    applies: false
     note:
   yagni:
-    applies:
+    applies: false
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      premised on the idea that CGI script writers may not anticipate the 
+      impact of certain inputs
```

Comment:
Lacking a subject of the sentence here, again perhaps a little more detail and a full sentence would add more value to the CVE